### PR TITLE
MLflow UI Sync

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/12026?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12026/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12026
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

WIP, double-checking to make sure things are good.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/server/js/src/common/utils/FeatureUtils.ts [hunk 1]
```diff
@@ -6,41 +6,43 @@
...
  */
 export const shouldDisableLegacyRunCompareCharts = () => false;
+
 /**
- * UI feature preview: displays data lineage (datasets used) in experiment runs
+ * Enables features related to deep learning: Phase 3
+ * Includes improved runs visibility controls, reworked hover tooltip and improved handling of charts on run details page.
  */
-export const shouldEnableExperimentDatasetTracking = () => true;
+export const shouldEnableDeepLearningUIPhase3 = () => false;
```
Comment (hubertzub-db) [inline mlflow/server/js/src/common/utils/FeatureUtils.ts:14]: I think we can enable this one already, WDYT @daniellok-db ?

File: mlflow/server/js/src/experiment-tracking/METADATA [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# go/metadata
```
Comment (hubertzub-db) [inline mlflow/server/js/src/experiment-tracking/METADATA:1]: this file definitely should be excluded

File: mlflow/server/js/src/common/components/Prompt.tsx
Comment (hubertzub-db) [inline mlflow/server/js/src/common/components/Prompt.tsx:1]: it seems that the `block()` function is missing in this particular react-router version brought from Databricks :/
Can you fix line #23 to be

```
const unblock = block?.(() => {
```
?

IMO lack of this feature is a minor regression, but we can fix it properly as a followup by finding proper react router version setup and make it specific to OSS

File: mlflow/server/js/src/common/utils/FeatureUtils.ts (hunk 1 above)
Comment (daniellok-db) [reply mlflow/server/js/src/common/utils/FeatureUtils.ts:14]: yeah let's do it

File: mlflow/server/js/src/experiment-tracking/components/runs-charts/hooks/useSampledMetricHistory.tsx [hunk 3]
```diff
@@ -110,7 +110,7 @@ export const useSampledMetricHistory = (params: {
 
     // Base fetching function, used for both initial call and subsequent auto-refresh calls
     const fetchMetricsFn = async (isAutoRefreshing = false) => {
-      const runUuids = runUuidsSerialized.split(',').filter((runUuid: string) => runUuid !== '');
+      const runUuids = runUuidsSerialized.split(',');
```
Comment (hubertzub-db) [inline mlflow/server/js/src/experiment-tracking/components/runs-charts/hooks/useSampledMetricHistory.tsx:113]: Can we leave this line with `.filter`? this is from reverted code, but is actually a useful fix. I can help with backsync conflicts if any

**Review by hubertzub-db — APPROVED** [review]
- did a lot of manual testing, overall everything works
- the code looks good

**Comment by daniellok-db** [comment]
@smurching would you mind taking a look at the changes in model registry?

**Review by smurching — COMMENTED** [review]
Thanks @daniellok-db for running the sync - I took a quick look at the registry code changes, they all look expected + reasonable - if Hubert's already QA'd the UI manually then LG to me to merge!

**Pull request merged**